GEN-257-Transaction-Simulation-Extension

gal-blockfence opened the pull request:
<img width="512" alt="image" src="https://user-images.githubusercontent.com/130892418/236675289-4dc4cc60-87f8-4254-97be-f53df81af2dc.png">

tal-blockfence commented on src/components/ContentDecoder.tsx:
```diff
@@ -34,6 +42,11 @@ export function ContentDecoder({ to, chainId = '1', descriptionResult, analyzeRe
                 url={url}
             />
             <Styled.Results>
+                {analyzeResult.transaction_simulation && shouldShowSimulation(analyzeResult.transaction_simulation) && (
```
nit: two checks feel redundant, include 'analyzeResult.transaction_simulation' in shouldShowSimulation

tal-blockfence commented on src/types/api.ts:
```diff
@@ -30,6 +30,7 @@ export type DataEnrichment = {
 export type Stat = {
     name: string;
     value: string;
+    icon?: string;
```
unused

tal-blockfence commented on src/components/Simulation.tsx:
```diff
@@ -0,0 +1,37 @@
+}
+
+export function Simulation({ simulation, defaultState = false }: SimulationProps) {
+    const logo = <img src={'../assets/logo.png'} width='24' />;
```
lets change the icon, something like this perhaps : https://iconscout.com/unicons/explore/line?redirect=%2Funicons%2Fexplore%2Fline%3Fcolor%3D%2525230092E4%26name%3Dexchange-alt

tal-blockfence commented on src/shared/api.tsx:
```diff
@@ -9,7 +9,10 @@ async function _fetchFunction<ResponseType>(
     to: string,
-    url: string
+    url: string,
+    from = '',
```
nit: we can skip the default values

tal-blockfence commented on src/components/Simulation.tsx:
```diff
@@ -0,0 +1,37 @@
+    defaultState?: boolean;
+}
+
+const shouldShowTransaction = (transaction: SimulatedTransaction) => {
```
lets show the empty transaction with '0' instead of skipping it

gal-blockfence commented:
Need some refinement:
<img width="512" alt="Screenshot 2023-05-08 at 11 06 07" src="https://user-images.githubusercontent.com/130892418/236770759-9a37b05d-8bed-429f-9315-9d5051b35f6c.png">

omri-blockfence reviewed (changes requested):
1. Add currency for the gas (on Ethereum Mainnet it's wei or gwei (1M wei = 1 gwei). Better to write it as short as possible (so use gwei instead of showing a long number, when possible). For other supported networks I think it is the same, but please verify (https://www.alchemy.com/gwei-calculator#:~:text=There%20are%201%20billion%20wei,1%2C000%2C000%2C000%20gwei%20in%20one%20ether.) 
2. Use less numbers after the decimal point for the tokens

eyal-blockfence commented on src/components/Simulation.tsx:
```diff
@@ -0,0 +1,44 @@
+            <Styled.Container>
+                <Styled.SectionContainer>
+                    <Styled.Bold>
+                        {"Out"}
```
```js
<Styled.Bold>Out</Styled.Bold>
```

eyal-blockfence commented on src/components/Simulation.tsx:
```diff
@@ -0,0 +1,44 @@
+                    </Styled.CenteredIconWithText>
+                </Styled.SectionContainer>
+            </Styled.Container>
+            {simulation?.gas_used && "Gas used: " + parseInt(simulation.gas_used)}
```
```js
{simulation?.gas_used && `Gas used: ${simulation.gas_used}`
```
It would be better to wrap in an element (like div or something)
```js
{simulation?.gas_used && <Styled.Gas>Gas used: {simulation.gas_used}</Styled.Gas>}
```

eyal-blockfence commented on src/components/Simulation.tsx:
```diff
@@ -0,0 +1,44 @@
+export function Simulation({ simulation, defaultState = false }: SimulationProps) {
+    return (
+        
+        <Collapsable title={'Transaction Simulation'} icon={<UilExchangeAlt />} defaultState={defaultState}>            
```
`title="Transaction Simulation"`

eyal-blockfence commented on src/components/Simulation.tsx:
```diff
@@ -0,0 +1,44 @@
+}
+
+const getAmountString = (amount?: string, symbol?: string) => {
+    return amount ? amount + " " + symbol : "0";
```
```js
    return amount ? `${amount} ${symbol}` : "0";
```

gal-blockfence commented:
Addressed comments and added $ values:
<img width="512" alt="Screenshot 2023-05-09 at 17 23 15" src="https://github.com/blockfence-io/extension/assets/130892418/89d9966b-f140-4652-a809-a54e80af481e">

tal-blockfence commented:
reminder - hide the gas price until the price is calculated
